### PR TITLE
STYLE: enable ruff TCH on some file

### PR DESCRIPTION
**Pull request by luke396**
xref #51740, to fix
```
"pandas/core/construction.py" = ["TCH"] 
"pandas/core/missing.py" = ["TCH"] 
"pandas/tseries/*" = ["TCH"] 
"pandas/tests/*" = ["TCH"] 
"pandas/plotting/*" = ["TCH"] 
```

### CODE REVIEW

File: pandas/tseries/__init__.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# ruff: noqa: TCH004
```
Comment (luke396) [inline pandas/tseries/__init__.py:1]: same as #51812

File: pandas/tests/extension/date/array.py [hunk 2]
```diff
@@ -10,10 +10,8 @@
 
 import numpy as np
 
-from pandas._typing import (
-    Dtype,
-    PositionalIndexer,
-)
+from pandas._typing import Dtype  # noqa: TCH001
+from pandas._typing import PositionalIndexer  # noqa: TCH001
```
Comment (luke396) [inline pandas/tests/extension/date/array.py:14]: if add `TYPE_CHECKING`, will cause import error, in `pandas/tests/io/json/test_json_table_schema_ext_dtype.py`

**Comment by MarcoGorelli** [comment]
thanks for your pr, looks like there's some errors https://github.com/pandas-dev/pandas/actions/runs/4362414904/jobs/7627283459

**Comment by luke396** [comment]
> thanks for your pr, looks like there's some errors https://github.com/pandas-dev/pandas/actions/runs/4362414904/jobs/7627283459

This was because of a careless mistake and should be fine now

File: pandas/tests/extension/date/array.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/tests/extension/date/array.py:14]: any idea why? would it work to add `from __future__ import annotations` at the top of this file?

Comment (twoertwein) [reply pandas/tests/extension/date/array.py:14]: Based on the pytest failures, adding a future import to `pandas/tests/extension/date/array.py` should fix this. (Wasn't there a script to enforce that all files have future imports?)

Comment (luke396) [reply pandas/tests/extension/date/array.py]: It seems like there have been a lot of changes to the type annotations, and I can't determine whether they will have a significant impact or not.

Comment (luke396) [reply pandas/tests/extension/date/array.py]: > It seems like there have been a lot of changes to the type annotations, and I can't determine whether they will have a significant impact or not.
> 
> Edit: if we add a script to enforce all files have future imports, the type annotation maybe change a lot too.

I searched for some information and found that these type annotation changes might be due to a new annotation feature introduced in a newer version of Python. It seems that this change does not have a significant impact, except for compatibility with older versions of Python.

**Review by twoertwein — APPROVED** [review]

File: pandas/tests/extension/date/array.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/tests/extension/date/array.py]: the check excludes test files (though perhaps it should include them?)

https://github.com/pandas-dev/pandas/blob/f7f0df109f84ec70a61718b285b73a7527e65c91/.pre-commit-config.yaml#L402-L413

**Review by MarcoGorelli — APPROVED** [review]
thanks @luke396 ! not many left, we're getting there

File: pandas/tests/extension/date/array.py (hunk 2 above)
Comment (luke396) [reply pandas/tests/extension/date/array.py]: > the check excludes test files (though perhaps it should include them?)
> 
> https://github.com/pandas-dev/pandas/blob/f7f0df109f84ec70a61718b285b73a7527e65c91/.pre-commit-config.yaml#L402-L413

Sorry for the late reply. If necessary, I open a pr and can manually or automatically add 'from future import annotations'.

**Pull request merged**